openfe/setup/protein_molecule.py

```diff
@@ -0,0 +1,59 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+from rdkit import Chem
+
+from openfe.utils.typing import RDKitMol, OEMol
+
+
+class ProteinMolecule:
+    """Wrapper around a Protein representation
+
+    This representation is immutable.  If you want to make any modifications,
+    do this in an appropriate toolkit then remake this class.
+    """
+    def __init__(self, rdkit: RDKitMol, name=""):
+        self._rdkit = rdkit
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    @classmethod
+    def from_pdbfile(cls, pdbfile: str, name=""):
+        m = Chem.MolFromPDBFile(pdbfile)
+        if m is None:
+            raise ValueError(f"RDKit failed to produce a molecule from "
+                             "{pdbfile}")
+        return cls(rdkit=m, name=name)
+
+    @classmethod
+    def from_pdbxfile(cls, pdbxfile: str, name=""):
```

Comment:
Do we have a long term strategy for this? I know you were trying to implement this @richardjgowers but I can't remember if it got anywhere.